wire-ios/Wire-iOS UnitTests/ImageMetadataTests.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+@testable import Wire
+
+final class ImageMetadataTests: XCTestCase {
+
+    func testImageMetadata() throws {
+        // checking if the original image fits the requirements for this test
+        
+        let originalImageData = imageData(resource: "test_image_with_gps_metadata", fileExtension: "jpg")
+        let originalImageMetadata = imageMetadata(originalImageData)
+        
+        XCTAssert(!originalImageMetadata.isEmpty) // the original image should have metadata
+        XCTAssert(metadataHasGps(originalImageMetadata)) // ... including GPS metadata
+        
+        // checking the image data that has been converted to UIImage
+        
+        let uiImage = UIImage(data: originalImageData)!
+        
+        // converted to jpeg
+        
+        let uiImageJpegData = uiImage.jpegData(compressionQuality: 1.0)
+        let uiImageJpegConvertedMetadata = imageMetadata(uiImageJpegData!)
+        
+        XCTAssert(!uiImageJpegConvertedMetadata.isEmpty) // the process of conversion to UIImage jpeg should have kept some metadata
+        XCTAssert(!metadataHasGps(uiImageJpegConvertedMetadata)) // ... but the GPS metadata should be gone
+        
+        // converted to png
+        
+        let uiImagePngData = uiImage.pngData()
+        let uiImagePngConvertedMetadata = imageMetadata(uiImagePngData!)
+        
+        XCTAssert(!uiImagePngConvertedMetadata.isEmpty) // the process of conversion to UIImage png should have kept some metadata
+        XCTAssert(!metadataHasGps(uiImagePngConvertedMetadata)) // ... but the GPS metadata should be gone
+        
+        // checking if there is no metadata after trying to remove the metadata
+        
+        let imageDataWithRemovedMetadata = try! originalImageData.wr_removingImageMetadata()
+        XCTAssert(imageMetadata(imageDataWithRemovedMetadata).isEmpty) // no metadata available anymore
```

Comment:
suggeestion: add comment to the message failure test instead
```suggestion
        XCTAssert(imageMetadata(imageDataWithRemovedMetadata).isEmpty, "no metadata available anymore")
```